### PR TITLE
  Allow overlapping ranges in case of single stack IPv6.

### PR DESCRIPTION
**Pull request by axel7born**
**How to categorize this PR?**

/area networking
/kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Allow overlapping network ranges in case of single stack IPv6.
```

### CODE REVIEW

File: pkg/apis/extensions/validation/network_test.go [hunk 1]
```diff
@@ -163,16 +163,6 @@ var _ = Describe("Network validation tests", func() {
 				}))))
 			})
```
Comment (ScheererJ) [inline pkg/apis/extensions/validation/network_test.go:165]: Please remove empty line

**Review by ScheererJ — APPROVED** [review]
Thanks for allowing the new setup with joined node, pod and service networks, which simplifies IPv6 networking. I just found a minor nit.

/lgtm
/approve

**Review by DockToFuture — APPROVED** [review]
/lgtm

File: pkg/apis/extensions/validation/network_test.go [hunk 2]
```diff
@@ -162,17 +162,6 @@ var _ = Describe("Network validation tests", func() {
 					"Field": Equal("spec.serviceCIDR"),
 				}))))
 			})
-
-			It("should forbid Network with overlapping pod and service CIDRs", func() {
```
Comment (ialidzhikov) [inline pkg/apis/extensions/validation/network_test.go:166]: Add a test case that overlapping ranges are allowed for IPv6 (validation is passing for them)?

/hold

File: pkg/apis/extensions/validation/network_test.go [hunk 3]
```diff
@@ -168,10 +168,7 @@ var _ = Describe("Network validation tests", func() {
 				network.Spec.PodCIDR = network.Spec.ServiceCIDR
 
 				errorList := ValidateNetwork(network)
-				Expect(errorList).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
```
Comment (ialidzhikov) [inline pkg/apis/extensions/validation/network_test.go:171]: L166:

```diff
- 			It("should forbid Network with overlapping pod and service CIDRs", func() {
+ 			It("should allow Network with overlapping pod and service CIDRs", func() {
```

**Review by ialidzhikov — COMMENTED** [review]
/lgtm

**Pull request merged**